codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -334,18 +335,16 @@ public final int numChildren() {
         }
 
         @Override
-        public final Collection<? extends Http2Stream> children() {
-            return children.values();
-        }
-
-        @Override
-        public final boolean hasChild(int streamId) {
-            return child(streamId) != null;
-        }
-
-        @Override
-        public final Http2Stream child(int streamId) {
-            return children.get(streamId);
+        public Http2Stream forEachChild(Http2StreamVisitor visitor) throws Http2Exception {
+            Http2Stream resultStream = null;
+            for (IntObjectHashMap.Entry<DefaultStream> entry : children.entries()) {
```

Comment:
Yeah that was the original idea, but TBH I hadn't put much thought into it since I was thinking we would replace the map with a more easily iterated collection.